wordpress/wp-content/plugins/memberful-wp/src/acl.php

```diff
@@ -195,7 +214,135 @@ function memberful_wp_extract_slug_ids_and_user($args) {
  * @param integer $post_id ID of the post that should have access checked
  */
 function memberful_can_user_access_post( $user, $post ) {
-  $restricted_posts = memberful_wp_user_disallowed_post_ids( $user );
+  $cache_result = wp_cache_get( "user-{$user}--post-{$post}", "memberful_post_access", $found = false );
```

Comment:
Is there a performance benefit to this change? The benchmarks showed that post protection with terms wasn't slower than the original implementation. Also, this adds additional complexity and we need to take care of cache invalidation which I don't see anywhere. Is it present?

Anyways, I would vote against adding caching in this PR. It's pretty complex even without caching so if we want to introduce caching for checking permission, then we should probably do it in a separate pull request.

Reply:
Also, I don't think that this actually works as expected. `$found` should be passed as a reference, however, it should be the fourth argument not third. I'm pretty sure that here we just set `$found = false` and pass `false` as the third argument with the fourth argument being empty.